### PR TITLE
CMAKE: Add source_group() CMake code to adjust how the plugins are presented in Visual Studio

### PR DESCRIPTION
**Pull request by imikejackson**

### CODE REVIEW

File: src/PipelineRunner/CMakeLists.txt [hunk 1]
```diff
@@ -12,4 +12,10 @@ target_sources(PipelineRunner
...
 
-target_link_libraries(PipelineRunner complex::complex)
\ No newline at end of file
+target_link_libraries(PipelineRunner complex::complex)
+
+source_group("PiplineRunner" FILES     
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/PipelineRunner.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/PRObserver.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/PRObserver.hpp
+)
```
Comment (JDuffeyBQ) [inline src/PipelineRunner/CMakeLists.txt:21]: Add newline for end of file.

**Review by JDuffeyBQ — CHANGES_REQUESTED** [review]

File: src/PipelineRunner/CMakeLists.txt [hunk 2]
```diff
@@ -12,4 +12,10 @@ target_sources(PipelineRunner
...
     )
 
-target_link_libraries(PipelineRunner complex::complex)
\ No newline at end of file
+target_link_libraries(PipelineRunner complex::complex)
+
+source_group("PiplineRunner" FILES     
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/PipelineRunner.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/PRObserver.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/PRObserver.hpp
```
[the remark this answers is not in the document]
Comment (imikejackson) [reply src/PipelineRunner/CMakeLists.txt]: Done

File: src/PipelineRunner/CMakeLists.txt (hunk 1 above)
Comment (imikejackson) [reply src/PipelineRunner/CMakeLists.txt]: Done

File: cmake/Plugin.cmake [hunk 3]
```diff
@@ -165,6 +165,8 @@ function(create_complex_plugin)
     )
   endforeach()
 
+  source_group("${NAME}/Filters" FILES )
+
```
[the remark this answers is not in the document]
Comment (imikejackson) [reply cmake/Plugin.cmake]: Fixed

File: cmake/Plugin.cmake [hunk 4]
```diff
@@ -251,7 +251,8 @@ function(create_complex_plugin)
       ${${ARGS_NAME}_Plugin_SRCS}
   )
 
-  source_group(TREE "${${ARGS_NAME}_SOURCE_DIR}/src/${ARGS_NAME}" PREFIX ${ARGS_NAME} FILES ${${ARGS_NAME}_ARGS_FILTER_FILES})
+  source_group(TREE "${${ARGS_NAME}_SOURCE_DIR}/src/${ARGS_NAME}" PREFIX ${ARGS_NAME} FILES ${${ARGS_NAME}_Plugin_HDRS} ${${ARGS_NAME}_Plugin_SRCS})
+  source_group("Generated" FILES ${${ARGS_NAME}_GENERATED_HEADERS} )
```
Comment (JDuffeyBQ) [inline cmake/Plugin.cmake:255]:
```suggestion
  source_group("Generated" FILES ${${ARGS_NAME}_GENERATED_HEADERS})
```

File: src/PipelineRunner/CMakeLists.txt [hunk 5]
```diff
@@ -3,13 +3,25 @@ project(PipelineRunner
...
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/PipelineRunner.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/PRObserver.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/PRObserver.hpp
+    ${PipelineRunner_HDRS}
+    ${PipelineRunner_SRCS}
     )
 
-target_link_libraries(PipelineRunner complex::complex)
\ No newline at end of file
+target_link_libraries(PipelineRunner complex::complex)
```
Comment (JDuffeyBQ) [inline src/PipelineRunner/CMakeLists.txt:24]:
```suggestion
target_link_libraries(PipelineRunner PRIVATE complex::complex)
```

**Review by JDuffeyBQ — CHANGES_REQUESTED** [review]

**Review by JDuffeyBQ — APPROVED** [review]

**Pull request merged**